Adding Missing Files and Implemented fix for TYPO3 Version

ResonanceBlade opened the pull request:
I added the missing files and implemented a fix for TYPO3 Version > 6.1

felixhasse committed:
Adding Missing Files and Implemented fix for TYPO3 Version

apfelbox commented on ext_emconf.php:
```diff
@@ -0,0 +1,44 @@
+
+$EM_CONF[$_EXTKEY] = array(
+	'title' => 'TYPO3 version checker',
+	'description' => 'Returns TYPO3 Version as JSON',
```
"Returns the TYPO3 version ..."

apfelbox commented on ext_localconf.php:
```diff
@@ -0,0 +1,17 @@
+    }
+}
+
+$TYPO3_CONF_VARS['FE']['eID_include']['becklynversionchecker'] = 'EXT:becklyn_version_checker/class.BecklynVersionChecker.php';
```
In this file, you shouldn't create a global variable. Instead you should move the logic to determine the version into your extension class.

apfelbox commented on ext_localconf.php:
```diff
@@ -0,0 +1,17 @@
+    if(class_exists('t3lib_utility_VersionNumber')) { 
+        $GLOBALS['TYPO3_VERSION'] = t3lib_utility_VersionNumber::convertVersionNumberToInteger(TYPO3_version);
+    } // TYPO3 Version 6.1.x
+    elseif(class_exists(\TYPO3\CMS\Core\Utility\VersionNumberUtility)) {
```
If you don't pass it in as string, you are actually already using this class. This will produce a fatal error if the class does not exist.

felixhasse committed:
Implemented TYPO3_version constants and removed the workaround

Implementation of the TYPO3_version constants instead of the deprecated $GLOBALS variant and removed the workaround.

apfelbox commented on class.BecklynVersionChecker.php:
```diff
@@ -0,0 +1,29 @@
+        echo json_encode(
+            array(
+                'system' => "typo3",
+                'version' => TYPO3_version
```
Seit welcher Version gibt es diese Konstante?

ResonanceBlade replied:
Gibt es schon seit der TYPO3 3.x
